rust/lance-linalg/build.rs

```diff
@@ -48,7 +52,11 @@ fn main() {
         {
             // It's likely the compiler doesn't support the sapphirerapids architecture
             // Clang 12 and GCC 11 are the first versions with sapphire rapids support
-            eprintln!("Skipping Sapphire Rapids build due to error: {}", err);
+            println!(
+                "cargo:warning=Skipping build of optimized f16 kernels due to error: {}",
+                err
+            );
+            return;
```

Comment:
I don't think we want to return here. There is a window of `clang < 12` but `> 6` where we have f16 support but not sapphire rapids. Maybe we can have an outer check?

Reply:
Also, if the build fails later, we can give a hint: "Use Clang => 6 or GCC >= 12"

Reply:
We have a fallback for older GCC versions and GCC 11 is still the default on Ubuntu 22 (which is quite prevalent).  So I don't think we want to force GCC 12 to be a minimum.  I'll add the outer check.

Reply:
Now it tries to build the AVX512 version first.  If that fails a warning is printed (cannot build *optimized* kernels) and it tries to build the fallback version.  If that fails then it asks the user to upgrade gcc/clang.

Reply:
Ok, third time is the charm.  We now fail the build if `fp16kernels` feature is present but it cannot build the kernels.  In the error we give the user the choice of upgrading gcc/clang or removing the feature.